design/20190708.certificate-request-crd.md

```diff
@@ -206,6 +206,17 @@ ownership of the default pool of issuers in the cert-manager project.
 Until the mutating webhook is fully implemented, we will handle defaulting
 internally in the controller.
 
+### CertificateRequest Annotations
+In order for `CertificateRequest` controllers to resolve requests, extra
+information may be needed that is not present in the API Spec. To pass on this
+information, a set of one or more annotations should be defined, with reliable
+value pairs. These are;
+
+- `certmanager.k8s.io/private-key-secret-name`: The name of the secret, in the
+  same namespace as the `CertificateRequest`, that stores the private key which
+  was used to sign the x509 certificate signing request. This is required by the
+ `SelfSigning` issuer to sign its own certificate.
+
```

Comment:
We should phrase this very carefully to note that these annotations are *optional* and may not exist on CertificateRequest resources, so any implementation that intends to use them *must* handle the case where they are *not* specified, and either fallback gracefully or correctly mark the CertificateRequest as failed with a suitable reason/message description.

We should also make sure we make it clear that the *Certificates* controller *does* add this annotation, however other implementations may not.

Reply:
:+1:  makes sense